libindy/src/services/pool/transaction_handler.rs

```diff
@@ -288,37 +271,67 @@ impl TransactionHandler {
         }
     }
 
-    fn parse_reply_for_proof_checking(json_msg: &SJsonValue)
-                                      -> Option<(&str, &str, Vec<u8>, Option<String>)> {
-        trace!("TransactionHandler::parse_reply_for_proof_checking: >>> json_msg: {:?}", json_msg);
-
-        let xtype = if let Some(xtype) = json_msg["type"].as_str() {
-            trace!("TransactionHandler::parse_reply_for_proof_checking: xtype: {:?}", xtype);
-            xtype
+    fn parse_generic_reply_for_proof_checking(&self, json_msg: &SJsonValue, raw_msg: &str) -> Option<Vec<ParsedSP>> {
+        let type_ = if let Some(type_) = json_msg["type"].as_str() {
+            trace!("TransactionHandler::parse_generic_reply_for_proof_checking: type_: {:?}", type_);
+            type_
         } else {
-            trace!("TransactionHandler::parse_reply_for_proof_checking: <<< No type field");
+            debug!("TransactionHandler::parse_generic_reply_for_proof_checking: <<< No type field");
             return None;
         };
 
-        if !REQUESTS_FOR_STATE_PROOFS.contains(&xtype) {
-            //TODO GET_DDO, GET_TXN
-            trace!("TransactionHandler::parse_reply_for_proof_checking: <<< type not supported");
-            return None;
+        if REQUESTS_FOR_STATE_PROOFS.contains(&type_) {
+            trace!("TransactionHandler::parse_generic_reply_for_proof_checking: built-in");
+            TransactionHandler::parse_reply_for_builtin_sp(json_msg, type_)
```

Comment:
> TransactionHandler::parse_reply_for_builtin_sp(json_msg, type_)

I don't like this approach. I prefer to have n dedicated functions that will be called for each transaction type without any distinction between "embedded" and "external" sp parser.

Reply:
I prefer to do it in separate PR...